### PR TITLE
Add 4bit quantizer to onnx runtime doc

### PR DESCRIPTION
**Pull request by fajin-corp**
### Description
Introduce how to use matmul_4bits_quantizer to do weight only quantization.

### Motivation and Context
Add 4bit quantizer to onnx runtime doc

### CODE REVIEW

File: docs/performance/model-optimizations/quantization.md [hunk 1]
```diff
@@ -231,6 +231,57 @@ ONNX Runtime leverages the TensorRT Execution Provider for quantization on GPU n
...
+- [MatMul](https://github.com/onnx/onnx/blob/main/docs/Operators.md#matmul):
+  - The node is quantized only if the input `B` is constant
+  - support QOperator or QDQ format.
+  - If QOperator is selected, the node is converted to a [MatMulNBits](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftmatmulnbits) node. Weight `B` is blockwise quantized and saved in the new node. [HQQ](https://arxiv.org/pdf/2309.15531.pdf), [GPTQ](https://huggingface.co/docs/transformers/main/en/quantization/gptq) and RTN (default) algorithms are supported.
+  - If QDQ is selected, the MatMul node is replaced by a DequantizeLinear -> MatMul pair. Weight `B` is blockwise quantized and saved in the DequantizeLinear node as an initializer.
+- [Gather](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Gather):
+  - The node is quantized only if the input `data` is constant.
+  - support QOperator
+  - Gather is replaced by a [GatherBlockQuantized](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftgatherblockquantized) node. Input `data` is blockwise quantized and saved in the new node. Only support RTN algorithm.
```
Comment (yufenglee) [inline docs/performance/model-optimizations/quantization.md:245]: nit: change "replaced" to "quantized"

File: docs/performance/model-optimizations/quantization.md [hunk 2]
```diff
@@ -231,6 +231,57 @@ ONNX Runtime leverages the TensorRT Execution Provider for quantization on GPU n
...
+  model, 
+  nodes_to_exclude=None, # specify a list of nodes to exclude from quantizaiton
+  nodes_to_include=None, # specify a list of nodes to force include from quantization
+  algo_config=quant_config,)
+quant.process()
+quant.model.save_model_to_file(
+  model_int4_path,
+  True) # save data to external file
+
+```
```
Comment (yufenglee) [inline docs/performance/model-optimizations/quantization.md:283]: Could you please add a reference to "https://github.com/microsoft/onnxruntime-genai/tree/main/src/python/py/models#quantized-pytorch-model" for AWQ and GTPQ quantization?

File: docs/performance/model-optimizations/quantization.md (hunk 1 above)
Comment (fajin-corp) [reply docs/performance/model-optimizations/quantization.md]: updated

File: docs/performance/model-optimizations/quantization.md (hunk 2 above)
Comment (fajin-corp) [reply docs/performance/model-optimizations/quantization.md:283]: updated doc

**Review by yufenglee — APPROVED** [review]

**Pull request merged**